packages/android_alarm_manager/android/src/main/java/io/flutter/plugins/androidalarmmanager/AndroidAlarmManagerPlugin.java

```diff
@@ -47,54 +54,71 @@
    * connected to, and running against, the associated Flutter execution context.
    */
   public static void registerWith(Registrar registrar) {
-    // alarmManagerPluginChannel is the channel responsible for receiving the following messages
+    final AndroidAlarmManagerPlugin plugin = new AndroidAlarmManagerPlugin();
+    // Listen for FlutterView destruction so that this plugin can move itself
+    // to background mode.
+    registrar.addViewDestroyListener(plugin);
```

Comment:
@bkonyi do you know what the analogous behavior of this plugin is when using the new embedding? I assume that what this plugin is doing when the view is destroyed is important, right? If so, we won't be able to act on view destruction, but plugins will now have the opportunity to take action when the activity disappears, which is probably what this view destruction really signified in the first place...

Reply:
+1

Reply:
Actually, looking at this code now I'm uncertain as to what it even accomplishes. It appears that on view destruction we try and store the view in `AlarmService`... but we also create a view in `AlarmService.startBackgroundIsolate`. I _think_ we're safe to remove this... I'll try and see if it causes any issues.